### PR TITLE
Add trainee report FE

### PR DESCRIPTION
**Pull request by bibixx**
## What part of the app is affected?
* [x] Frontend
* [ ] Backend
* [ ] Infrastructure

### CODE REVIEW

File: backend/src/trainees/trainees.service.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { FindOptionsWhere, Repository } from 'typeorm';
 import { Transactional } from 'typeorm-transactional-cls-hooked';
+import { DriversLicenseCategoriesService } from '../drivers-license-category/drivers-license-category.service';
 // import { CourseReport } from '../course-reports/entities/course-report.entity';
```
Comment (maonat) [inline backend/src/trainees/trainees.service.ts:6]: Comments to be removed perhaps? @bibixx

File: backend/src/trainees/trainees.service.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { Injectable } from '@nestjs/common';
...
 import { OrganizationDomainService } from '../organization-domain/organization-domain.service';
 // import { ReportEntriesService } from '../report-entries/report-entries.service';
```
Comment (maonat) [inline backend/src/trainees/trainees.service.ts:8]: Comments to be removed perhaps? @bibixx

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -56,6 +58,7 @@ export const App = () => {
             <Route path="nowy" element={<TraineeNew />} />
             <Route path=":traineeId" element={<TraineeDetails />} />
             <Route path=":traineeId/edytuj" element={<TraineeEdit />} />
+            <Route path=":traineeId/raport" element={<TraineeReport />} />
           </Route>
           <Route
             path="/instruktorzy"
```
Comment (maonat) [inline frontend/src/App.tsx:64]: Do we want to keep english or polish paths? @bibixx

File: frontend/src/hooks/useCourseReportData/useCourseReportData.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
+import { GetCourseReportResponseDto } from '@osk/shared';
+import { useEffect, useReducer } from 'react';
+import useSWR from 'swr';
+import { ReportGroup, RowData } from '../../components/Report/Report';
+import { mapCourseReportDtoToReportGroups } from '../../components/Report/Report.utils';
+import { assertNever } from '../../utils/asserNever';
+
+// App State
```
Comment (maonat) [inline frontend/src/hooks/useCourseReportData/useCourseReportData.ts:8]: Is this comment required? @bibixx

File: frontend/src/hooks/useCourseReportData/useCourseReportData.ts [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  groups: ReportGroup[];
+  courseReportId: number;
+}
+
+type CourseReportDataState =
+  | CourseReportDataLoadingState
+  | CourseReportDataErrorState
+  | CourseReportDataDoneState;
+
+// App Actions
```
Comment (maonat) [inline frontend/src/hooks/useCourseReportData/useCourseReportData.ts:28]: Comment required? @bibixx

File: frontend/src/views/Trainees/TraineeNew/TraineeNew.tsx [hunk 6]
```diff
@@ -33,7 +33,7 @@ export const TraineeNew = () => {
         pesel: newTrainee.pesel || null,
         pkk: newTrainee.pkk,
         driversLicenseNumber: newTrainee.driversLicenseNumber || null,
-        driversLicenseCategoryId: 6, // TODO CHANGE THIS TO GET IT FROM BE FROM DRIVER LICENSE CATEGORY
+        driversLicenseCategoryId: newTrainee.driversLicenseCategory,
```
Comment (maonat) [inline frontend/src/views/Trainees/TraineeNew/TraineeNew.tsx:36]: Thanks! I forgot about this one. @bibixx

**Review by maonat — CHANGES_REQUESTED** [review]
There might be some comments to be removed on the FE and some in the BE 👍 @bibixx

File: frontend/src/hooks/useCourseReportData/useCourseReportData.ts (hunk 4 above)
Comment (bibixx) [reply frontend/src/hooks/useCourseReportData/useCourseReportData.ts:8]: Yup, I'm marking sections for types. All of the types below this one up until `App Actions` are types related to the app's state 😉

File: backend/src/trainees/trainees.service.ts (hunk 2 above)
Comment (bibixx) [reply backend/src/trainees/trainees.service.ts:8]: Let's resolve those in your (PR)[https://github.com/twoje-osk/twoje-osk/pull/78] if needed. Ideally this PR shouldn't contain any BE changes

**Pull request merged**